astropy/timeseries/periodograms/base.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from astropy.timeseries import TimeSeries, BinnedTimeSeries
+
+__all__ = ['BasePeriodogram']
+
+
+class BasePeriodogram:
+
+    @classmethod
+    def from_timeseries(cls, timeseries, *, column=None, error=None, **kwargs):
+        """
+        Initialize a periodogram from a time series object.
+
+        If a binned time series is passed, the time at the center of the bins is
+        used. Also note that this method automatically gets rid of NaN/undefined
+        values when initalizing the periodogram.
+
+        Parameters
+        ----------
+        column : str, optional
+            The name of the column containing the y values to use.
+        error : str or float or `~astropy.units.Quantity`, optional
```

Comment:
"error" -> "uncertainty"?

and as above, in the description below perhaps reword "y values" to something more like "signal"?

Reply:
Fixed